valuable-serde/tests/test.rs

```diff
@@ -0,0 +1,97 @@
+use std::collections::HashMap;
+
+use valuable::*;
+use valuable_serde::SerdeValue;
+
+#[derive(Valuable, serde::Serialize)]
+struct HelloWorld {
+    hello: &'static str,
+    world: World,
+    list: Vec<World>,
+    map: HashMap<&'static str, u32>,
+    bool_: bool,
+    unit: (),
+    enums: Vec<Enum>,
+}
+
+#[derive(Valuable, serde::Serialize)]
+struct World {
+    answer: usize,
+}
+
+#[derive(Valuable, serde::Serialize)]
+enum Enum {
+    Struct { x: &'static str },
+    Newtype(u8),
+    Tuple(u8, u16),
+    // TODO
+    // Unit,
```

Comment:
Currently, `Valuable` treats the unit variant/struct as an empty tuple variant/struct. Therefore, I'm not sure if we can provide a behavior that is fully consistent with serde on unit variant/struct.

Reply:
This seems related to #21, maybe provide some thoughts there?

Reply:
The problem I encountered here is mainly related to the unit struct, not to tuples. But in any case, we need to think about how to represent tuples...

https://github.com/tokio-rs/valuable/blob/9a68fc31ef5fdd3bee25d0cf59d437bd26ee81ee/valuable-serde/tests/test.rs#L85-L101